core/src/main/java/org/infinispan/persistence/util/PersistenceManagerCloseableSupplier.java

```diff
@@ -126,7 +132,7 @@ public void processEntry(MarshalledEntry<K, V> marshalledEntry, AdvancedCacheLoa
          closeLock.lock();
          try {
             // If is possible that someone inserted a value and then acquired the close lock - thus we must recheck
-            if (closed || (entry = queue.poll()) != null) {
+            if ((entry = queue.poll()) != null || closed) {
```

Comment:
I stared for 5 minutes at this code yesterday without realizing what switching the evaluation order would do...
I saw it when I looked again today, but may I suggest not mixing complex `if` conditions and assignments? :)